lib/metadata_listener/report/extracted_text.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'faraday'
+
+module MetadataListener
+  module Report
+    class ExtractedText
+      def self.call(**args)
+        new(args[:path], args[:endpoint], args[:api_token]).send
+      end
+
+      attr_reader :path, :endpoint, :api_token
+
+      def initialize(path, endpoint, api_token)
+        @path = path
+        @endpoint = endpoint
+        @api_token = api_token
+      end
+
+      def send
+        report_results if endpoint.present?
+      end
+
+      private
+
+        def report_results
+          response = connection.put do |req|
+            req.body = { derivatives: { text: params } }.to_json
+          end
+          MetadataListener.logger.warn("Report failed: #{response.body}") unless response.success?
+        end
+
+        def params
+          {
+            id: uploaded_file.split('/').last,
+            storage: uploaded_file.split('/').first,
```

Comment:
we're kind of using 'store', 'storage', and 'prefix' interchangeably throughout the codebase, and I fear it will cause some confusion in the future. 

the store is not always equal to the prefix, in our case it is but that may not always be the case. perhaps we code this in as store: 'derivatives', and in other places of the codebase where we're using `store` instead of `prefix` name it as such. The changes i'm pondering should probably be part of another refactor, however.

Reply:
The params here, id and storage, get passed to Scholarsphere directly to create the derivative in Shrine: 

```
file = Shrine.uploaded_file(id: 'asdf.txt', storage: 'derivatives', metadata: { ... })
file_reource.file_attacher.merge_derivatives(text: file)
```

But it doesn't have to be that way. We'd need to define what the interface is in Scholarsphere between metadata-listener and Shrine.

Reply:
Maybe SS should pass to MDL the "storage" maybe something like "{ storage: "store": derivatives", "prefix": "derivatives" } that way if it changes on one end, it automatically gets picked up on the other? I'm also kind of thinking out loud

Reply:
I don't think there' anything needed other than just the prefix at this point, but we should be able to do that at least. I've created #51 to capture all of that as well as some earlier issues with the MDL-SS link.